pkg/api/types.go

```diff
@@ -601,6 +601,12 @@ type EmptyDirVolumeSource struct {
 	// The default is "" which means to use the node's default medium.
 	// +optional
 	Medium StorageMedium
+	// Total amount of local storage required for this directory.
+	// The size limit is also applicable for Memory type.
+	// The default is nil which means that the limit is undefined.
+	// More info: http://kubernetes.io/docs/user-guide/volumes#emptydir
+	// +optional
+	SizeLimit resource.Quantity
```

Comment:
I want to understand how this field will work with memory backed volumes.

For example, if a pod has a memory request for 500Mi of memory, and has a memory backed volume of 1Gi, the total memory request is still 500Mi, correct?  Users are responsible for ensuring that memory backed volumes are sized as part of their pod requests similar to how /dev/shm works?  In addition, a BestEffort pod can have a memory backed volume that has a size limit without changing the pod qos?